### PR TITLE
[DFT][Examples] Make the oneMKL DFT example consistent with the other domains

### PR DESCRIPTION
**Pull request by s-Nick**
# Description

This patch updates the oneMKL DFT examples making them consistent with other domains. In particular, it removes the old split examples for mklcpu and mklgpu and it adds a new example that uses mklcpu and cuFFT backend, same backend combination used in other examples. It enables run-time example for mklcpu backend and it updates example README.

## All Submissions

- [x] Do all unit tests pass locally? Attach a log.
[dft_examples.txt](https://github.com/user-attachments/files/15897128/dft_examples.txt)

- [x] Have you formatted the code using clang-format?

### CODE REVIEW

File: examples/dft/compile_time_dispatching/CMakeLists.txt [hunk 1]
```diff
@@ -18,31 +18,32 @@
...
-endif()
-
-if(ENABLE_MKLCPU_BACKEND)
-  list(APPEND DFTI_CT_BACKENDS "mklcpu")
+set(DFT_CT_SOURCES "")
+if (ENABLE_MKLCPU_BACKEND AND ENABLE_CUFFT_BACKEND)
+  list(APPEND DFT_CT_SOURCES "complex_fwd_usm_mklcpu_cufft")
 endif()
 
-include(WarningsUtils)
```
Comment (Rbiessy) [inline examples/dft/compile_time_dispatching/CMakeLists.txt:31]: Please keep this and `onemkl_warnings` for the DFT domain. This is a difference I'd like to keep for the DFT domain and eventually we could add the same warnings to the other domains.

File: examples/README.md [hunk 2]
```diff
@@ -354,34 +353,33 @@ Random number generator example with uniform distribution ran OK on MKLCPU and C
 
 ## dft
 
-Compile-time dispatching example with MKLGPU backend
```
Comment (Rbiessy) [inline examples/README.md]: Note we won't need to update the example output once https://github.com/oneapi-src/oneMKL/pull/515 is merged

File: examples/dft/compile_time_dispatching/CMakeLists.txt (hunk 1 above)
Comment (s-Nick) [reply examples/dft/compile_time_dispatching/CMakeLists.txt:31]: Thank for your review @Rbiessy I added it back in 4468702

File: examples/README.md (hunk 2 above)
Comment (s-Nick) [reply examples/README.md]: Since it will be removed in any case, I reverted the commit and I would not change it in this PR. Addressed in cd54f10

**Review by Rbiessy — APPROVED** [review]

File: examples/README.md [hunk 3]
```diff
@@ -3,15 +3,14 @@ oneAPI Math Kernel Library (oneMKL) Interfaces offers examples with the followin
 - blas: level3/gemm_usm  
 - rng: uniform_usm  
 - lapack: getrs_usm
-- dft: complex_fwd_buffer, real_fwd_usm
+- dft: complex_fwd_usm, real_fwd_usm
```
Comment (mkrainiuk) [inline examples/README.md:6]: Any specific reasons for switching from SYCL Buffer APU to USM?

**Review by mkrainiuk — APPROVED** [review]
Thank you, looks good to me, I have one minor question.

File: examples/README.md (hunk 3 above)
Comment (s-Nick) [reply examples/README.md:6]: Thank for your review @mkrainiuk 
The change is just for consistency with other domains

**Pull request merged**